### DISCUSSION TITLE
merge control flow edges?

### DISCUSSION THREAD

**Discussion by yufengzjj** [Vector35/binaryninja-api#1704] [General] [answered]
I'm doing some research on control flow/data flow.
I noticed that if one basic block got too many incoming edges it looks really bad.

is there any options or any plans to merge the incoming edges? like IDA did

<img width="395" alt="9D33D722-D9F8-4249-9D93-3F33E6222534" src="https://user-images.githubusercontent.com/7138035/82757524-8c156100-9e13-11ea-892a-678468ce1dff.png">

<img width="229" alt="7D05CA8F-16CA-4117-8EF7-3E3E50A3411F" src="https://user-images.githubusercontent.com/7138035/82757549-ba933c00-9e13-11ea-8eb1-f94f7d587944.png">

**Comment by plafosse** [comment, marked as answer]
This is one of the enhancements we hope to address when we complete this issue: https://github.com/Vector35/binaryninja-api/issues/58
